### PR TITLE
Refactor ssh

### PR DESCRIPTION
**Pull request by foolusion**
This is massive, maybe we can run through this together

### CODE REVIEW

**Comment by zachpuck** [comment]
LGTM. I have not completed any tests at this time though.

**Comment by foolusion** [comment]
I ran a test, it worked

File: pkg/ssh/ssh.go [hunk 1]
```diff
@@ -65,58 +72,43 @@ func NewClient(address, user string, privateKey []byte) (*Client, error) {
 	return &Client{c: c, Client: client}, nil
 }
 
-type CommandRunner struct {
-	Stdout     io.Writer
-	Stderr     io.Writer
-	err        error
-	currentCmd string
+type BatchRunner struct {
```
Comment (davidewatson) [inline pkg/ssh/ssh.go]: There are no internal uses of this type. Can we add comments to explain it's purpose?

Comment (foolusion) [reply pkg/ssh/ssh.go]: This one is actually replacing the command runner. It definitely needs a comment. But you can see how it's used in the pkg/controller/machine/ssh-commands.go file https://github.com/samsung-cnct/cma-ssh/pull/45/files#diff-9c30a87edbcefc660f992acf777fd4a2R123

**Pull request merged**